src/dins/core/information_service.py

```diff
@@ -31,53 +36,121 @@ class InformationService(InformationServicePort):
+            # Only log if information to be inserted is a mismatch
+            if existing_mapping != dataset_file_accessions:
+                await self._dataset_dao.update(dataset_file_accessions)
+                log.info("Updated existing dataset %s.", dataset_accession)
+        except ResourceNotFoundError:
+            await self._dataset_dao.insert(dataset_file_accessions)
+            log.debug(
+                "Successfully inserted file accession mapping for dataset %s.",
+                dataset_accession,
+            )
+
+    async def register_file_information(
+        self, file: event_schemas.FileInternallyRegistered
     ):
         """Store information for a file newly registered with the Internal File Registry."""
         file_information = FileInformation(
-            file_id=file_registered.file_id,
-            size=file_registered.decrypted_size,
-            sha256_hash=file_registered.decrypted_sha256,
+            accession=file.file_id,
+            size=file.decrypted_size,
+            sha256_hash=file.decrypted_sha256,
         )
-        file_id = file_information.file_id
+        file_id = file_information.accession
 
+        # inverted logic due to raw pymongo exception exposed by hexkit
         try:
             existing_information = await self._file_information_dao.get_by_id(
                 id_=file_id
             )
-            log.debug(f"Found existing information for file {file_id}")
+            log.debug("Found existing information for file %s.", file_id)
             # Only log if information to be inserted is a mismatch
             if existing_information != file_information:
-                information_exists = self.MismatchingInformationAlreadyRegistered(
+                information_exists = self.MismatchingFileInformationAlreadyRegistered(
                     file_id=file_id
                 )
                 log.error(information_exists)
         except ResourceNotFoundError:
             await self._file_information_dao.insert(file_information)
-            log.debug(f"Successfully inserted information for file {file_id}")
+            log.debug("Successfully inserted information for file %s.", file_id)
+
+    async def serve_dataset_information(
+        self, dataset_id: str
+    ) -> DatasetFileInformation:
+        """Retrieve stored public information for the given dataset ID to be served by the API."""
+        try:
+            dataset = await self._dataset_dao.get_by_id(dataset_id)
+            log.debug("Found mapping for dataset %s.", dataset_id)
+        except ResourceNotFoundError as error:
+            dataset_not_found = self.DatasetNotFoundError(dataset_accession=dataset_id)
+            log.warning(dataset_not_found)
+            raise dataset_not_found from error
+
+        file_information: list[FileAccession | FileInformation] = []
+
+        for file_accession in sorted(dataset.file_accessions):
```

Comment:
The file accessions field is already a list and should therefore have a definite sequence. Did you run into problems with it returning different sequences here?

Reply:
Not sure if the accessions in the incoming event that populates the database are always sorted. 
Wanted to make sure they always will be for this endpoint.